packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -131,7 +132,7 @@ export function mediaUpload( {
 
 		// Set temporary URL to create placeholder media file, this is replaced
 		// with final file from media gallery when upload is `done` below
-		filesSet.push( { url: window.URL.createObjectURL( mediaFile ) } );
+		filesSet.push( { url: createBlobURL( mediaFile ) } );
```

Comment:
Each call to createBlobURL should eventually have a revokeBlobURL to make sure the file used by the blob is removed from the memory when not need anymore. In this case, revokeBlobURL should be called as soon as createMediaFromFile finishes with success or fails.

Reply:
Added with 80ebf4a 👍